metrics/prometheus.go

```diff
@@ -97,6 +99,12 @@ type PrometheusCollector struct {
 	containerMetrics    []containerMetric
 	containerLabelsFunc ContainerLabelsFunc
 	includedMetrics     container.MetricSet
+
+	// custom options to pass to stats collection routine. The option will be populated
+	// once in the first scrape so the scrape request should be the same each time (as
+	// usual with prometheus scrapes should already be in most cases)
+	updateOnce sync.Once
```

Comment:
Is it not possible to do this per-request?  It seems like this would have strange behavior if there are multiple callers.

Reply:
`Gatherer` interface only has method `Collect(ch)` which didnt accept any argument, and can be called concurrently, so do this per-request is hard to do with a singleton collector as we have now, as we would have to lock the request.
However, i looked at it again, and dont think this is a good idea any more, mainly because this prevents user from changing maxAge by changing prometheus configuration.

Alternative would be the previous approach in #2000 which is to create a collector per request ( like how node_exporter currently does it https://github.com/prometheus/node_exporter/blob/master/node_exporter.go#L96-L137 ). Actually now that i think about it abit more, that does seem like a better approach that what we have now, as registrry and collector are quite lightweight, and the collection is done infrequently.